components/size_aware_image.jsx

```diff
@@ -170,6 +196,79 @@ export default class SizeAwareImage extends React.PureComponent {
             />
         );
 
+        // copyLink, download are two buttons overlayed on image preview
+        // copyLinkTooltip, downloadTooltip are tooltips for the buttons respectively.
+        // if linkCopiedRecently is true, defaultMessage would be 'Copy Link', else 'Copied!'
+
+        const copyLinkTooltip = (
+            <Tooltip
+                id='copy-link-tooltip'
+                className='hidden-xs'
+            >
+                <FormattedMessage
+                    id='single_image_view.copy_link_tooltip'
+                    defaultMessage={this.state.linkCopiedRecently ? 'Copied' : 'Copy link'}
+                />
+            </Tooltip>
+        );
+        const copyLink = (
+            <OverlayTrigger
+                className='hidden-xs'
+                delayShow={500}
+                placement='top'
+                overlay={copyLinkTooltip}
+                rootClose={true}
+            >
+                <button
+                    className={classNames('style--none', 'size-aware-image__copy_link', {
+                        'size-aware-image__copy_link--recently_copied': this.state.linkCopiedRecently,
+                    })}
+                    aria-label={`${ariaLabelImage}${localizeMessage('copy url')}`}
+                    onClick={this.copyLinkToAsset}
+                >
+                    {this.state.linkCopiedRecently ? (
+                        <i className='icon icon-check style--none'/>
+                    ) : (
+                        <i className='icon icon-link-variant style--none'/>
+                    )}
+                </button>
+            </OverlayTrigger>
+        );
+
+        const downloadTooltip = (
+            <Tooltip
+                id='download-preview-tooltip'
+                className='hidden-xs'
+            >
+                <FormattedMessage
+                    id='single_image_view.download_tooltip'
+                    defaultMessage='Download'
+                />
+            </Tooltip>
+        );
+
+        const download = (
+            <OverlayTrigger
+                className='hidden-xs'
+                delayShow={500}
+                placement='top'
+                overlay={downloadTooltip}
+                rootClose={true}
+            >
+                <a
+                    href={this.isInternalImage() ? fileURL : src}
+                    className='style--none size-aware-image__download'
+                    target='_blank'
+                    rel='noopener noreferrer'
+                    download={ariaLabelImage}
+                    aria-label={`${ariaLabelImage}${localizeMessage('download image')}`}
+                >
+                    <i className='icon icon-download-outline style--none'/>
```

Comment:
nit (since this works as well): We can use icon components from the `@mattermost/compass-icons` now (renders plain `svg`), which are a lot easier to align correctly.
example: 

```jsx
import {DownloadOutlineIcon} from '@mattermost/compass-icons/components'

...

<DownloadOutlineIcon
    size={24} // any number will be use to set width and height
    color={'#FFF'} // any valid CSS color string (even variables are possible)
/>
```

All other props will be spread onto the `svg` as well, so using aria attributes do work.